features/networking/sriov.feature

```diff
@@ -66,6 +69,806 @@ Feature: Sriov related scenarios
+    """
+    When I run the :get admin command with:
+      | resource  | net-attach-def        |
+      | namespace | <%= cb.usr_project%>  |
+    Then the step should succeed
+    And the output should contain "mlx277-netdevice"
+    """
+    And I use the "<%= cb.usr_project%>" project
+    Given I obtain test data file "networking/sriov/pod/sriov-macvlan.yaml"
+    When I run oc create over "sriov-macvlan.yaml" replacing paths:
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"] | mlx277-netdevice |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | name=sriov-macvlan |
+    When I execute on the pod:
+      | bash | -c | cat /sys/class/net/net1/address |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :pod1_net1_mac clipboard
+    Given I use the "<%= pod.node_name %>" node
+    And I run commands on the host:
+      | ip link show ens2f0 |
+    Then the step should succeed
+    And the output should contain "<%= cb.pod1_net1_mac%>"
+
+  # @author zzhao@redhat.com
+  # @case_id OCP-24776
+  @destructive
+  @admin
+  Scenario: The default sriov networkpolicy should be able to restore by sriov operator when it was deleted	
+    Given the sriov operator is running well
+    When I run the :delete admin command with:
+      | object_type       | sriovnetworknodepolicies |
+      | object_name_or_id | default                  |
+    Then the step should succeed
+    When I run the :get admin command with:
+      | resource    |  sriovnetworknodepolicies |
+    Then the step should succeed
+    And the output should contain "default"        
+    Given I obtain test data file "networking/sriov/sriovnetworkpolicy/mlx277-netdevice.yaml"
+    Given I create sriov resource with following:
+      | cr_yaml       | mlx277-netdevice.yaml    |
+      | cr_name       | mlx277-netdevice         |
+      | resource_type | sriovnetworknodepolicies |
+    Then the step should succeed
+    And I wait up to 500 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource    | sriovnetworknodestates           |
+      | namespace   | openshift-sriov-network-operator |
+      | o           | yaml                             |
+    Then the step should succeed
+    And the output should contain:
+      | mlx277-netdevice      |
+      | syncStatus: Succeeded |
+      | vfID: 1               |
+    """
+
+  # @author zzhao@redhat.com
+  @destructive
+  @admin
+  Scenario Outline: SR-IOV resource can be disable by edit SR-IOV Operator Config
+    Given the sriov operator is running well
+    Given <sriov-feature> is disabled
+    #check the related resource should be removed
+    And I wait up to 30 seconds for the steps to pass:
+    """
+    When I run the :get admin command with:
+      | resource    | ds |
+    Then the step should succeed
+    And the output should not contain "<keyword>"    
```

Comment:
Given /^the( admin)? (\w+) named #{QUOTED} does not exist(?: in the#{OPT_QUOTED} project)?$/ do |who, resource_type, resource_name, project_name|

Reply:
updated